xsd-core/src/main/java/io/ballerina/xsd/core/visitor/VisitorUtils.java

```diff
@@ -0,0 +1,225 @@
+                    ? OPEN_SQUARE_BRACKET + EMPTY_STRING + CLOSE_SQUARE_BRACKET + WHITESPACE : EMPTY_STRING
+            );
+        }
+    }
+
+    public static void handleFixedValues(StringBuilder builder, Node typeNode, Node fixedNode) {
+        builder.append(fixedNode != null
+                ? generateFixedValue(deriveType(typeNode), fixedNode.getNodeValue())
+                : deriveType(typeNode)).append(WHITESPACE);
+    }
+
+    public static String deriveType(Node node) {
+        String derivedType = node.getNodeValue().contains(COLON) ?
+                node.getNodeValue().substring(node.getNodeValue().indexOf(COLON) + 1) : node.getNodeValue();
+        return typeGenerator(derivedType);
+    }
+
+    public static String typeGenerator(String typeName) {
+        switch (typeName) {
+            case TIME, DATE_TIME, DATE, G_YEAR_MONTH, G_YEAR, STRING, LANGUAGE, DURATION, ANY_URI -> {
+                return STRING;
+            }
+            case INTEGER, LONG, NEGATIVE_INTEGER, NON_POSITIVE_INTEGER, POSITIVE_INTEGER, SHORT,
+                    UNSIGNED_LONG, UNSIGNED_INT, UNSIGNED_SHORT, UNSIGNED_BYTE, INT, NON_NEGATIVE_INTEGER -> {
+                return INT;
+            }
+            case BASE64_BINARY -> {
+                return BYTE_ARRAY;
+            }
+            default -> {
+                return typeName;
+            }
+        }
+    }
+
+    public static boolean isSimpleType(String type) {
+        String typeName = type.contains(COLON) ? type.substring(type.indexOf(COLON) + 1) : type;
+        String[] simpleTypes = {
+                TIME, DATE_TIME, DATE, G_YEAR_MONTH, G_YEAR, STRING, LANGUAGE, DURATION,
+                INTEGER, LONG, NEGATIVE_INTEGER, NON_POSITIVE_INTEGER, POSITIVE_INTEGER, SHORT,
+                UNSIGNED_LONG, UNSIGNED_INT, UNSIGNED_SHORT,
+                UNSIGNED_BYTE, INT, BASE64_BINARY, BOOLEAN, FLOAT, DOUBLE, DECIMAL, ANY_URI, NON_NEGATIVE_INTEGER
+        };
+        return Arrays.stream(simpleTypes).toList().contains(typeName);
+    }
+
+    public static Iterable<Node> asIterable(NodeList nodeList) {
+        return () -> new Iterator<>() {
+            private int index = 0;
+
+            @Override
+            public boolean hasNext() {
+                return index < nodeList.getLength();
+            }
+
+            @Override
+            public Node next() {
+                if (!hasNext()) {
+                    throw new NoSuchElementException();
+                }
+                return nodeList.item(index++);
+            }
+        };
+    }
+
+    public static String convertToCamelCase(String pascalCase) {
+        if (pascalCase == null || pascalCase.isEmpty()) {
+            return pascalCase;
+        }
+        return Character.toLowerCase(pascalCase.charAt(0)) + pascalCase.substring(1);
+    }
+
+    public static String sanitizeString(String input) {
+        if (!input.matches(".*[!@$%^&*()_\\-|/\\\\\\s\\d].*")
```

Comment:
imo, define this in class level with a proper var name as a constants will be improve the readability

Reply:
Fixed with 43067b6